src/biokbase/narrative/tests/test_jobcomm.py

```diff
@@ -1225,30 +1328,352 @@ def test_translate_require_job_id(self):
+        with self.assertRaisesRegex(JobIDException, JOBS_NOT_PROVIDED_ERR):
+            rqs[0].job_id_list
 
         rq_msg = make_comm_msg(JobRequest.REQUIRE_JOB_ID[0], ["a", "b"], False)
         rqs = JobRequest.translate(rq_msg)
         self.assertEqual(len(rqs), 2)
         self.assertEqual(rqs[0].job_id, "a")
-        self.assertEqual(rqs[0].job_id_list, None)
+        with self.assertRaisesRegex(JobIDException, JOBS_NOT_PROVIDED_ERR):
+            rqs[0].job_id_list
         self.assertEqual(rqs[1].job_id, "b")
-        self.assertEqual(rqs[1].job_id_list, None)
+        with self.assertRaisesRegex(JobIDException, JOBS_NOT_PROVIDED_ERR):
+            rqs[1].job_id_list
 
     def test_translate_require_job_id_list(self):
         rq_msg = make_comm_msg(JobRequest.REQUIRE_JOB_ID_LIST[0], "a", False)
         rqs = JobRequest.translate(rq_msg)
         self.assertEqual(len(rqs), 1)
-        self.assertEqual(rqs[0].job_id, None)
+        with self.assertRaisesRegex(JobIDException, JOB_NOT_PROVIDED_ERR):
+            rqs[0].job_id
         self.assertEqual(rqs[0].job_id_list, ["a"])
 
         rq_msg = make_comm_msg(JobRequest.REQUIRE_JOB_ID_LIST[0], ["a", "b"], False)
         rqs = JobRequest.translate(rq_msg)
         self.assertEqual(len(rqs), 1)
-        self.assertEqual(rqs[0].job_id, None)
+        with self.assertRaisesRegex(JobIDException, JOB_NOT_PROVIDED_ERR):
+            rqs[0].job_id
         self.assertEqual(rqs[0].job_id_list, ["a", "b"])
 
     def test_translate_doesnt_require_any_job_ids(self):
         rq_msg = make_comm_msg("all_status", None, False)
         rqs = JobRequest.translate(rq_msg)
         self.assertEqual(len(rqs), 1)
+
+    def test_translate_both_inputs(self):
+        msg = {
+            "msg_id": "some_id",
+            "content": {"data": {
+                "request_type": "job_status",
+                "job_id": "ababab",
+                "job_id_list": []
+            }},
+        }
+        with self.assertRaisesRegex(ValueError, "Both job_id and job_id_list present"):
+            JobRequest.translate(msg)
+
+
+class exc_to_msgTestCase(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.maxDiff = None
+        cls.jc = JobComm()
+        cls.jc._comm = MockComm()
+
+    def setUp(self):
+        self.jc._comm.clear_message_cache()
+
+    @staticmethod
+    def foo(req, f):
+        with exc_to_msg(req):
+            f()
+
+    @staticmethod
+    def bar(req, f, indic):
+        with exc_to_msg(req):
+            try:
+                f()
+            except Exception:
+                indic += ["A"]
+                raise
+            indic += ["B"]
+        indic += ["C"]
+
+    def test_with_nested_try__raise(self):
+        job_id_list = [BATCH_TERMINATED, JOB_TERMINATED]
+        req_type = "stop_job_status_loop"
+        req = make_comm_msg(req_type, job_id_list, True)
+        message = (
+            "Casteism is the investment in keeping the hierarchy "
```

Comment:
I get that hierarchies can be useful, but I'm not sure that Job states need to follow a caste system... 😅